keystone_performance/load_test_multiple_users.py

```diff
@@ -0,0 +1,419 @@
+        if args.project_id:
+            project_info = {'id': args.project_id}
+        else:
+            project_info = {'name': args.project_name}
+        if args.project_domain_id:
+            project_info['domain'] = {'id': args.project_domain_id}
+        else:
+            project_info['domain'] = {'name': args.project_domain_name}
+        auth_req_body = {
+            'auth': {
+                'identity': {
+                    'methods': ['password'],
+                    'password': {
+                        'user': {
+                            'name': args.username,
+                            'domain': user_domain_info,
+                            'password': args.password
+                        }
+                    }
+                },
+                'scope': {
+                    'project': project_info
+                }
+            }
+        }
+
+        headers = {'content-type': 'application/json'}
+        r = requests.post(auth_url, json=auth_req_body, headers=headers)
+        if r.status_code != 201:
+            raise Exception('Error: Failed to get keystone token for getting ' +
+                'user info. Status code: ' + str(r.status_code))
+        token = r.headers['x-subject-token']
+
+        domains_url = '%s/v3/domains' % args.url
+        params = {'name': 'Default'}
+        headers = {'X-Auth-Token': token}
+        r = requests.get(domains_url, params=params, headers=headers)
+        if r.status_code != 200:
+            raise Exception('Error: Failed to get domain list. Status code: ' +
+                str(r.status_code))
+        domain_id = r.json()['domains'][0]['id']
+
+        users_url = '%s/v3/users' % args.url
+        params = {'domain_id': domain_id, 'name__startswith': 'load_user'}
+        r = requests.get(users_url, params=params, headers=headers)
+        if r.status_code != 200:
+            raise Exception('Error: Failed to get user list. Status code: ' +
+                str(r.status_code))
+        users = r.json()['users']
+
+        if len(users) == 0:
+            raise Exception('Error: No test users found with names '
+                'starting with load_user')
+
+        self._user_infos = []
+        for user in users:
+            self._user_infos.append({'username': user['name'],
+                'domain_id': user['domain_id']})
+
+    def get_random_user_info(self):
+        rand_index = random.randrange(0, len(self._user_infos) - 1)
+        return self._user_infos[rand_index]
+
+
+class TestTracker(object):
+    def __init__(self, args):
+        self._args = args
+        self._user_info = UserInfo(args)
+
+        self._agent = client.Agent(reactor)
+
+        if args.type == 'quick':
+            self._run_time = 15  # seconds
+            self._concurrencies = [1, 2, 4, 8]
+        else:  # Full run
+            self._run_time = 60  # seconds
+            self._concurrencies = [1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, ]
```

Comment:
It's not configurable. The goal is to not have thousands of configuration options but instead to keep the code simple.